src/commands/lambda/functions/instrument.ts

```diff
@@ -267,47 +257,72 @@ export const calculateUpdateRequest = async (
     const lambdaLibraryLayerName = LAYER_LOOKUP[runtime as LayerKey]
     let fullLambdaLibraryLayerARN: string | undefined
     if (settings.layerVersion !== undefined || settings.interactive) {
-      let layerVersion = settings.layerVersion
+      layerOrTraceVersion = settings.layerVersion
       if (settings.interactive && !settings.layerVersion) {
-        layerVersion = await findLatestLayerVersion(config.Runtime as LayerKey, region)
+        layerOrTraceVersion = await findLatestLayerVersion(config.Runtime as LayerKey, region)
       }
-      fullLambdaLibraryLayerARN = `${lambdaLibraryLayerArn}:${layerVersion}`
+      fullLambdaLibraryLayerARN = `${lambdaLibraryLayerArn}:${layerOrTraceVersion}`
     }
     layerARNs = addLayerArn(fullLambdaLibraryLayerARN, lambdaLibraryLayerName, layerARNs)
   }
 
   const lambdaExtensionLayerArn = getLayerArn(config, EXTENSION_LAYER_KEY as LayerKey, region, settings)
   let fullExtensionLayerARN: string | undefined
+  let extensionVersion: number | undefined
   if (settings.extensionVersion !== undefined || settings.interactive) {
-    let extensionVersion = settings.extensionVersion
+    extensionVersion = settings.extensionVersion
     if (settings.interactive && !settings.extensionVersion) {
       extensionVersion = await findLatestLayerVersion(EXTENSION_LAYER_KEY as LayerKey, region)
     }
     fullExtensionLayerARN = `${lambdaExtensionLayerArn}:${extensionVersion}`
   }
-
   layerARNs = addLayerArn(fullExtensionLayerARN, DD_LAMBDA_EXTENSION_LAYER_NAME, layerARNs)
 
+  // Special handling for .NET and Java to support universal instrumentation
+  if (runtimeType === RuntimeType.DOTNET || runtimeType === RuntimeType.JAVA) {
+    if (layerOrTraceVersion && isExtensionSupportUniversalInstrumentation(runtimeType, extensionVersion)) {
+      // If the user configures the trace version and the extension support univeral instrumenation
+      // Then check whether the trace and extension are compatible with each other
+      if (isTracerCompatibleWithExtension(runtimeType, layerOrTraceVersion)) {
+        needsUpdate = true
+        newEnvVars[AWS_LAMBDA_EXEC_WRAPPER_VAR] = AWS_LAMBDA_EXEC_WRAPPER
+      } else {
+        throw new Error(
+          `For the ${runtime} runtime, the dd-trace version ${layerOrTraceVersion} is not compatible with the dd-extension version ${extensionVersion}`
+        )
+      }
+    } else if (runtimeType === RuntimeType.DOTNET) {
+      // If it is an old extension version or the trace version is null, leave it is as the old workflow
+      if (
+        !isExtensionSupportUniversalInstrumentation(runtimeType, extensionVersion) &&
+        config.Architectures?.includes(ARM64_ARCHITECTURE)
+      ) {
+        throw new Error(
+          'Instrumenting arm64 architecture is not supported for the given dd-extension version. Please choose the latest dd-extension version or use x86_64 architecture.'
```

Comment:
Is this correct? I believe we released the ARM .NET versions of the layer. @IvanTopolcic  since Storms is out

Reply:
In the origin code, it throws an error for the combination of .NET and ARM64. My understanding is that our old tracer&extension versions do not support .NET running on ARM64. However, the latest version of extension and tracer supports.  My implementation was based on this understanding.

Reply:
Ahh, that makes sense